pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -179,9 +179,25 @@ func getTaskRunByName(t *testing.T, taskRuns map[string]*v1beta1.TaskRun, expect
 // It will fatal the test if an error occurred.
 func getTaskRunsForPipelineRun(ctx context.Context, t *testing.T, clients test.Clients, namespace string, prName string) map[string]*v1beta1.TaskRun {
 	t.Helper()
+	labelSelector := pipeline.PipelineRunLabelKey + "=" + prName
+	return getTaskRuns(ctx, t, clients, namespace, labelSelector)
+}
+
+// getTaskRunsForPipelineTask returns the set of TaskRuns associated with the input PipelineRun and PipelineTask
+// It will fatal the test if an error occurred.
+func getTaskRunsForPipelineTask(ctx context.Context, t *testing.T, clients test.Clients, namespace string, prName string, ptLabel string) map[string]*v1beta1.TaskRun {
+	t.Helper()
+	labelSelector := pipeline.PipelineRunLabelKey + "=" + prName + "," + pipeline.PipelineTaskLabelKey + "=" + ptLabel
+	return getTaskRuns(ctx, t, clients, namespace, labelSelector)
+}
```

Comment:
nice!!

it'd be amazing if you could clean up the other tests to use this function in a follow up PR 🙏🏾